### PR TITLE
Implement profiles on MCU side

### LINKED ISSUE #47: Introduce profiles to save different configurations
Currently there is no way to save more than one configuration. Since it can be quite annoying to reconfigure all settings, it would be nice to store all settings in profiles so that they can be switched easily.

- [x] Implement profiles on MCU side
- [ ] Implement profiles on UI side

### PR DESCRIPTION
**Pull request by TheRealKasumi**
partially closes #47 

With this PR profiles are added to the MCU configuration. WiFi and Audio Unit settings as well as calibration data are excluded from this to avoid problems when switching profiles. All other settings are now stored in up to 50 profiles. Corresponding endpoints are implemented.

Also as always.... some general coding improvements that I found on the way.

### CODE REVIEW

File: mcu/include/configuration/Configuration.h [hunk 1]
```diff
@@ -22,8 +22,10 @@
 #ifndef CONFIGURATION_H
 #define CONFIGURATION_H
 
+#include <stdint.h>
+#include <vector>
 #include <tuple>
-#include <Arduino.h>
```
Comment (JakobLichterfeld) [inline mcu/include/configuration/Configuration.h:26]: yiha, one not optimal dependency less

**Review by JakobLichterfeld — APPROVED** [review]
lgtm. Love the endpoint concept.

**Comment by danielbayerlein** [comment]
@TheRealKasumi Can you please update the swagger file?

**Comment by danielbayerlein** [comment]
UI: #148

**Review by danielbayerlein — DISMISSED** [review]

**Review by danielbayerlein — APPROVED** [review]

**Pull request merged**